src/aws-cpp-sdk-core/source/internal/AWSHttpResourceClient.cpp

```diff
@@ -441,6 +446,46 @@ namespace Aws
             static std::shared_ptr<EC2MetadataClient> s_ec2metadataClient(nullptr);
         #endif
 
+        void InitEC2MetadataClient(const Aws::Client::ClientConfiguration::CredentialProviderConfiguration& credentialConfig) {
+          if (s_ec2metadataClient)
+          {
+            return;
+          }
+          Aws::String ec2MetadataServiceEndpoint = Aws::Environment::GetEnv("AWS_EC2_METADATA_SERVICE_ENDPOINT");
```

Comment:
this is the same as the other overloaded `InitEC2MetadataClient` calculation for endpoint, we shouldnt be copy and pasting logic. please refactor out to a shared endpoint calculation function.